src/typings.ts

```diff
@@ -420,17 +420,35 @@ export interface Signature {
   concatSig: string
 }
 
-export interface IdentityWalletSchema {
-  TLWalletVersion: number
-  ethersKeystore: string
-  identityAddress: string
-  walletType: string
+// WALLET
+export type WalletTypeEthers = 'ethers'
+export type WalletTypeIdentity = 'identity'
+export type WalletType = WalletTypeEthers | WalletTypeIdentity
+
+export interface TLWalletSchema {
+  version: number
+  type: WalletType
+  meta?: any
+}
+
+export interface SigningKey {
+  privateKey: string
+  mnemonic: string
+  derivationPath: string
```

Comment:
Thinking about it, we dont have to store the derivation path, because we currently also do not support using in in recover from mnemonic. So we can just hard code it as we do it in recover from mnemonic.